src/status_im/contexts/profile/edit/modal/events.cljs

```diff
@@ -0,0 +1,56 @@
+(ns status-im.contexts.profile.edit.modal.events
+  (:require [status-im.contexts.profile.edit.introduce-yourself.view :as introduce-yourself]
+            [utils.re-frame :as rf]))
+
+(rf/reg-event-fx
+ :profile/edit-profile
+ (fn [_ [{:keys [display-name picture color on-success]}]]
+   {:fx (cond-> []
+          display-name
+          (conj [:dispatch
+                 [:profile/edit-name
+                  {:display-name   display-name
+                   :navigate-back? false
+                   :show-toast?    false}]])
+          color
+          (conj [:dispatch
+                 [:profile/edit-accent-colour
+                  {:color          color
+                   :navigate-back? false
+                   :show-toast?    false}]])
+          picture
+          (conj [:dispatch
+                 [:profile/edit-picture
+                  {:picture     picture
+                   :show-toast? false}]])
+
+          (nil? picture)
+          (conj [:dispatch [:profile/delete-picture {:show-toast? false}]])
+
+          :always
+          (conj [:dispatch-n [[:navigate-back] on-success]]))}))
```

Comment:
`:dispatch-n` was made redundant by `:fx`. You can use:

```clojure
(cond-> []
  :always
  (conj [:dispatch [:event-a]]
        [:dispatch [:event-b]]))
;; => [[:dispatch [:event-a]]
;;     [:dispatch [:event-b]]]
```

Reply:
Thanks for noticing!
Fixed!